test/functional/tests/io_class/test_io_class_file.py

```diff
@@ -364,53 +564,99 @@ def test_ioclass_file_size(filesystem):
     base_size = Size(random.randint(50, 1000) * 2, Unit.Blocks4096)
     size_to_class = {
         base_size: 1,
-        base_size - Unit.Blocks4096: 2,
-        base_size + Unit.Blocks4096: 3,
+        base_size - Size(1, Unit.Blocks4096): 2,
+        base_size + Size(1, Unit.Blocks4096): 3,
         base_size / 2: 4,
-        base_size / 2 - Unit.Blocks4096: 4,
-        base_size / 2 + Unit.Blocks4096: 2,
+        base_size / 2 - Size(1, Unit.Blocks4096): 4,
+        base_size / 2 + Size(1, Unit.Blocks4096): 2,
         base_size * 2: 5,
-        base_size * 2 - Unit.Blocks4096: 3,
-        base_size * 2 + Unit.Blocks4096: 5,
+        base_size * 2 - Size(1, Unit.Blocks4096): 3,
+        base_size * 2 + Size(1, Unit.Blocks4096): 5,
     }
 
-    with TestRun.step("Prepare cache and core."):
-        cache, core = prepare(default_allocation="1.00")
+    with TestRun.step("Prepare cache and core devices"):
+        ioclass_config.remove_ioclass_config(ioclass_config_path)
+
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(10, Unit.GibiByte)])
+        core_device.create_partitions([Size(20, Unit.GibiByte)])
+
+        cache_device = cache_device.partitions[0]
+        core_device = core_device.partitions[0]
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(
+            cache_device,
+            cache_mode=CacheMode.WB,
+            cache_line_size=CacheLineSize.LINE_4KiB,
+            force=True,
+        )
+
+    with TestRun.step("Disable udev"):
+        Udev.disable()
+
+    with TestRun.step("Disable cleaning policy and sequential cutoff"):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step("Create filesystem on core device"):
+        core_device.create_filesystem(fs_type=filesystem)
+
+    with TestRun.step("Add core"):
+        core = casadm.add_core(cache, core_dev=core_device)
+
+    with TestRun.step("Create and load IO class config"):
+        ioclass_config.create_ioclass_config(
+            add_default_rule=False, ioclass_config_path=ioclass_config_path
+        )
+        # Set all non-target workloads to pass-through mode to avoid caching anything
+        # except files matching IO classification rules(based on file size)
+
+        ioclass_config.create_ioclass_config(
+            add_default_rule=True, ioclass_config_path=ioclass_config_path
+        )
```

Comment:
1. we just need 1 IO class config creation (with the default rule)
2. you have a default class here which is allocated (although I'm not sure why), so it doesn't actually do what is written in the comment